### PR TITLE
Proposing to create a hello world with expo

### PR DESCRIPTION
**Pull request by vinils**
Reasons:
- less files, less bugs, support and maintenance
- As with react-native CLI (create-react-native-app), this can make it easier to people get involved and help/produce more
- Expo has many facilities, hot reload, publish and others

Related to https://github.com/Microsoft/reactxp/issues/208

### CODE REVIEW

File: samples/hello-world-expo/src/ProgressIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// /**
```
Comment (dryganets) [inline samples/hello-world-expo/src/ProgressIndicator.tsx]: Do we need in this commented code?

File: samples/hello-world-expo/src/SecondPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                        style={ styles.progressMargin }
+                        progress={ this.state.progressValue }
+                        fillColor={ '#ddd' }
+                        size={ 32 }
+                    /> */}
+
+                    <RX.Text style={ [styles.titleText, styles.videoTitleText] }>
+                        Here is a video using the Video extension
+                    </RX.Text>
+                    {/* <RXVideo
```
Comment (dryganets) [inline samples/hello-world-expo/src/SecondPanel.tsx]: I recently fixed video package.
If it was your main concern here - you could fix the PR accordingly.
See: https://github.com/Microsoft/reactxp/pull/229
for reference.

File: samples/hello-world-expo/src/ProgressIndicator.tsx (hunk 1 above)
Comment (vinils) [reply samples/hello-world-expo/src/ProgressIndicator.tsx]: No, can be removed!

File: samples/hello-world-expo/src/SecondPanel.tsx (hunk 2 above)
Comment (vinils) [reply samples/hello-world-expo/src/SecondPanel.tsx]: Done!

File: package.json [hunk 3]
```diff
@@ -2,7 +2,6 @@
   "name": "reactxp",
   "version": "0.42.0-rc.24",
   "description": "Cross-platform abstraction layer for writing React-based applications a single time that work identically across web, React Native, and Electron distribution",
-  "author": "ReactXP Team <reactxp@microsoft.com>",
```
Comment (dryganets) [inline package.json]: I think it should stay here.

File: samples/hello-world-expo/README.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+### Building for Web
+
+- Run `npm run web-watch`. This compiles the TypeScript code and recompiles it whenever any files are changed.
+- Open `index.html` in your browser to view the result.
+
+### Building for React Native
+
+Install the [Expo](https://expo.io) app on your iOS or Android phone, and use the QR code in the terminal to open your app. Find the QR scanner on the Projects tab of the app. When you're ready to share your project with others (for example, by deploying to an app store)
+
+- Run `exp start`
```
Comment (dryganets) [inline samples/hello-world-expo/README.md]: I try it and it says:
Error: Missing exp.json. See https://docs.expo.io/

probably something is missing?

File: samples/hello-world-expo/package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "desert-walk-expo-19",
+  "version": "0.0.0",
+  "description": "Hello Expo!",
+  "author": "Jan Aagaard <jan@aagaard.net>",
```
Comment (dryganets) [inline samples/hello-world-expo/package.json]: "author": "ReactXP Team <reactxp@microsoft.com>",

File: samples/hello-world-expo/package.json [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  "private": true,
+  "main": "node_modules/expo/AppEntry.js",
+  "scripts": {
+    "web-watch": "webpack --progress --colors --watch"
+  },
+  "dependencies": {
+    "expo": "19.0.1",
+    "react": "16.0.0-alpha.12",
+    "react-dom": "^16.0.0-alpha.12",
+    "react-native": "https://github.com/expo/react-native/archive/sdk-19.0.0.tar.gz",
```
Comment (dryganets) [inline samples/hello-world-expo/package.json]: Is it possible to use correct react-native version?

File: package.json (hunk 3 above)
Comment (vinils) [reply package.json]: done!

File: samples/hello-world-expo/App.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+// This entry point is request by expo. 
+// Shall be open an expo request to change this entry point to src/index.tsx
+// in order to get cross-platform and typescript standard
+import App from './src/App'
+
+export default App
```
[the remark this answers is not in the document]
Comment (vinils) [reply samples/hello-world-expo/App.js]: Done!

File: samples/hello-world-expo/package.json (hunk 5 above)
Comment (vinils) [reply samples/hello-world-expo/package.json]: Done!

File: samples/hello-world-expo/package.json (hunk 6 above)
Comment (vinils) [reply samples/hello-world-expo/package.json]: changed to     "react-native": "0.46.4",
I had some problems on <= 0.44 but newest CLI (expo+react-native) dont care that anymore. Not sure why, but changed!

**Comment by artalat** [comment]
Hey, how did you get it to work? I'm seeing #216 on `exp start`

**Comment by vinils** [comment]
@artalat I ran yarn install and Expo XDE. You can open an issue describing you problem on my repository and i´ll be glad to help you there!
@dryganets done!

**Comment by lagofvc** [comment]
@vinils, First, thanks for your work on this issue!   I agree with you:  Reactxp will gain more traction if expo can be used.   

I'm also trying to get the `hello-world-expo` sample to work from your Repo and I am also seeing #216 as @artalat mentioned.   I get : `undefined is not an object (evaluating 'Navigation.StateUtils')`.   

Happy to create an issue, but when I visit your repo I do not see the "Issues" tab available in order to create one for this error.   

Thanx again!

**Comment by aanari** [comment]
@vinils, Thank you for your hard work. I am having the same problem as @lagofvc, and would love to get this resolved so that your branch is working correctly.

Screenshot:

screen shot 2017-09-05 at 12 04 13 am

**Comment by vinils** [comment]
@lagofvc, @aanari there is an workarroudn for NavigatorExperimentalDelegate mentioned by @Fitzpasd
on https://github.com/Microsoft/reactxp/issues/216.

reproducing that here:

Workaround in the meantime: Don't use the Navigation API in reactxp, and comment out line 33 in node_modules/reactxp/dist/native-common/NavigatorExperimentalDelegate.js

...
var Types = require("../common/Types");
// var StateUtils = Navigation.StateUtils;
var NavigatorExperimentalDelegate = (function (_super) {
...

**Comment by lagofvc** [comment]
Awesome!! Thanx for the details @vinils.   That works!!

**Comment by erictraut** [comment]
I have mixed feelings about this submission. None of the maintainers of this repo use expo, and we've historically found that the sample code needs to be updated regularly when ReactXP changes (especially around major versions). I guess I'm OK with this sample being added as long as it is maintained by the broader community. If it becomes out of date and is not being maintained, I'll delete it in the future.

**Comment by vinils** [comment]
@erictraut I was waiting for this talk. 

First of all I would like to complement you for the great job to care about the direction of reactxp development and as far as you are taking expo in consideration I can see that you are really concerned about to take the best decision for reactxp.

I'll try to argue back and show some improve points i think you should try.

"I guess I'm OK with this sample being added as long as it is maintained by the broader community. If it becomes out of date and is not being maintained, I'll delete it in the future"

This submission will remove ios/android folders which means less files (less maintenance), easier community engagement and other stuffs that make me believe you guys should take in consideration to adopt expo or an extra work to maintain that.

No argue about to remove if is not being used.

"None of the maintainers of this repo use expo, and we've historically found that the sample code needs to be updated regularly when ReactXP changes (especially around major versions)"

In my point of view, I'd break sample's folder in a second problem (bigger one). Why reactxp should keep repeated files (such as gradlew, package.config, webpack, ios and others) since they don't "direct" belong  to reactxp project and are repeated for every sample?

My fix suggestion to this other problem is to create 2 CLI (one for expo and other for the actual one). remove those sample from the source and keep they on the docs/blog site and create automated test for those 3 samples running over those 2 CLI.

**Comment by vinils** [comment]
Any news? :)

**Comment by dryganets** [comment]
Hi, sorry had no time to figure out why expo fails on the video module. By some reason it makes ts look for another module typings in this module node_modules.

**Comment by erictraut** [comment]
No response to request for modifications in over a month, so closing the PR.

**Comment by evanrlong** [comment]
I have some additional info on this issue (I too was curious about using Expo with React XP).  I believe the problem with the video module is that reactxp-video uses react-native-video as its underlying engine.  The react-native-video library is not compatible with Expo by default since it uses native code instead of Javascript.  It only works if you eject your Expo project to Expokit and handle builds natively instead of through the expo app.  Since the main appeal (at least for me) of using Expo with React XP is to allow for development without XCode and Android Studio (for example to do ios app builds on Windows PCs), this isn't ideal.  

React XP could fix this problem by using the [video library](https://docs.expo.io/versions/latest/sdk/video.html) contained in Expo to drive the mobile component of reactxp-video instead of react-native-video.  However, as this forces React XP to add an Expo dependency, I'm sure that isn't a desirable solution.

This might be a good candidate for a new extension (something like reactxp-expo-video).  If i get time, I might take a crack at it.

**Comment by vinils** [comment]
@evanrlong if you have that time to make the crack, please share with me, the video is one of the pieces missing for my project

**Comment by evanrlong** [comment]
@vinils I certainly will.  It will probably have to wait until after the holidays.

**Pull request closed without merging**